app/controllers/publishers_controller.rb

```diff
@@ -396,37 +394,18 @@ def log_out
   def choose_new_channel_type
   end
 
-  def generate_statement
-    publisher = current_publisher
-    statement_period = params[:statement_period]
-    statement = PublisherStatementGenerator.new(publisher: publisher, statement_period: statement_period.to_sym).perform
-    SyncPublisherStatementJob.perform_later(publisher_statement_id: statement.id, send_email: true)
-    render(json: {
-      id: statement.id,
-      date: statement_period_date(statement.created_at),
-      period: statement_period_description(statement.period.to_sym)
-    }, status: 200)
-  end
-
-  def statement_ready
-    statement = PublisherStatement.find(params[:id])
-    if statement && statement.contents
-      head 204
-    else
-      head 404
-    end
-  end
-
   def statement
-    statement = PublisherStatement.find(params[:id])
+    statement_period = params[:statement_period]
+    @transactions = PublisherStatementGetter.new(publisher: current_publisher, statement_period: statement_period).perform
+    @statement_period = publisher_statement_period(@transactions)
```

Comment:
I prefer a `flash[:notice]` rather than opening a blank pdf based on the response from `PublisherStatementGetter.new`